### ISSUE TITLE
Unable to make basic example work

### ISSUE THREAD

**Issue by Todai88** [mosch/react-avatar-editor#263]
**PROBLEM:**

I'm trying to make React-Avatar work using React(1.6+) and TypeScript(latest).
I've been running into multiple issues through the implementation, as I can't seem to find a Typescript version of the JSX based on the [**demo**][1].

 
**Expected:**

I expect it to *just* work, when having typings for the project and its dependencies set up. I've had to manually change a bunch of things to make webpack build the package, but I can't make it work.

**Actual:**

Errors everywhere. Currently I'm getting this:
 

    Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
    
    Check the render method of `MyEditor`.
        in MyEditor
    
    Uncaught Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.

**Code:**

avatarUploader.tsx:

    import * as React from "react";
    import AvatarEditor from "react-avatar-editor"; 
    
    class MyEditor extends React.Component {
      render() {
        return (
          <AvatarEditor
            image={"http://example.com/initialimage.jpg"}
            width={250}
            height={250}
            border={50}
            color={[255, 255, 255, 0.6]} // RGBA
            scale={1.2}
            rotate={0}
          />
        )
      }
    }
    
    export default MyEditor

**fileItem.tsx:**

    import * as React from "react";
    import * as ReactDOM from "react-dom"; 
    import MyEditor from "./../avatarTest/avatarUploader" 
    
        ReactDOM.render(
            <MyEditor />
            ,  
            document.getElementById("body")
        );

**webpack.config.js:**

```
module.exports = {
    mode: "development",

    entry: {
        filemanager: "./src/fileItem/fileitem.tsx", 
        avatarTest: "./src/avatarTest/avatarUploader.tsx"
    },
    output: {
        filename: "[name]/[name].bundle.js",
        path: __dirname + "/dist/transpiled-components/"
    },

    watch: true,
    devtool: "source-map",

    resolve: {
        extensions: [".ts", ".tsx", ".js", ".json"]
    },

    module: {
        rules: [   
            { test: /\.ts$/, loader: "ts-loader" },
            { test: /\.tsx$/, loader: "babel-loader!ts-loader" }
        ],
    }, 
    externals: {
        "react": "React",
        "react-dom": "ReactDOM" 
    }
} 


```
  [1]: https://github.com/mosch/react-avatar-editor/tree/master/docs

**Comment by mosch** [comment]
Probably this component need's an update of the react version. I'll look into it.

**Comment by Jefski14** [comment]
Is there any news on this issue? I tried receating it in https://codesandbox.io/s/m5p9o1657p but there it works fine, but locally I'm not able to figure out what causes the problem. It's likely that is has something to do with the imports...

**Comment by Jefski14** [comment]
I got it to work using require, but i still got no idea on how to properly fix it.
`const AvatarEditor = require("react-avatar-editor")`

**Comment by 3xpedia** [comment]
Hello guys, same problem here.

The code : 

`
import React, {useState} from 'react';
import {Col, Row} from "react-bootstrap";
import {AvatarEditor} = from "react-avatar-editor";

const ImagePicker = props => {

    const [picture, setPicture] = useState(null);

    const changeFile = e => {
        if(e.target.files[0]) {
            let file = e.target.files[0];
            setPicture(file)
        }
    };

    return (
        <Row>
            <Col xs={12}>
                <input type="file" onChange={e => changeFile(e)}/>
            </Col>
            <Col xs={12}>
                {
                    picture && (
                        <AvatarEditor
                            image={picture}
                            width={250}
                            height={250}
                            border={50}
                            color={[255, 255, 255, 0.6]}
                            scale={1.2}
                            rotate={0}
                        />
                    )
                }
            </Col>
        </Row>
    )
};

export default ImagePicker
`

Basically a file input updating the state and an AvatarEditor rendering an image located in that state.
The error log is the same :
`Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.`

And changing the import for 
`
const AvatarEditor = require('react-avatar-editor');
`
Make the think OK.

Any idea on what is the problem or how to fix this?
